pkg/cpio/cpio.go

```diff
@@ -77,9 +78,28 @@ func (r Reader) ReadRecords() ([]Record, error) {
 
 type Writer struct {
 	rw RecordWriter
+	// There seems to be no harm done in stripping
+	// duplicate names when the record is written,
+	// and lots of harm done if we don't do it.
+	alreadyWritten map[string] bool
 }
 
 func (w Writer) WriteRecord(rec Record) error {
+	// we do NOT write records with absolute paths.
+	if filepath.IsAbs(rec.Name) {
```

Comment:
The other problem is that you really can't clean the path until it comes time to write the record. You won't be able to read it otherwise. IOW, abs paths are needed to create an archive but it's very, very desirable not to place them in the archive. I'm happy to see a better solution than this, I just am not sure I know what it is.